File: tests/src/main/java/se/ansman/kotshi/SomeEnum.java
package se.ansman.kotshi;

// Until https://youtrack.jetbrains.com/issue/KT-21433 is resolved this enum has to be written in Java
public enum SomeEnum {
    VALUE1,
    VALUE2,
    @JsonDefaultValue
    VALUE3,
    VALUE4,
    VALUE5
}

File: tests/src/main/kotlin/se/ansman/kotshi/SomeEnum.kt
package se.ansman.kotshi

import com.squareup.moshi.Json

@JsonSerializable
enum class SomeEnum {
    VALUE1,
    VALUE2,
    @OptIn(ExperimentalKotshiApi::class)
    @JsonProperty(name = "VALUE3-alt")
    VALUE3,
    @Json(name = "VALUE4-alt")
    VALUE4,
    VALUE5
}

@JsonSerializable
enum class SomeEnumWithFallback {
    VALUE1,
    VALUE2,
    @JsonDefaultValue
    VALUE3,
    VALUE4,
    VALUE5;
}
